hydra-main/src/main/java/com/addthis/hydra/job/spawn/balancer/SpawnBalancer.java

```diff
@@ -1403,15 +1403,38 @@ private JobTaskItemByHostMap generateTaskCountByHost(List<HostState> hosts, Iter
         return rv;
     }
 
-    private List<JobTaskMoveAssignment> pushTasksOntoHost(HostState host, Iterable<HostState> heavyHosts) {
+    private List<JobTaskMoveAssignment> pushTasksOntoHost(HostState host, Iterable<HostState> heavyHosts,
+                                                          boolean limitBytes, double byteLimitFactor, int moveLimit,
+                                                          boolean obeyDontAutobalanceMe) {
+
         MoveAssignmentList moveAssignments = new MoveAssignmentList(spawn, taskSizer);
+        List<JobTaskMoveAssignment> rv = purgeMisplacedTasks(host, moveLimit);
+        moveAssignments.addAll(rv);
+
+        // Available bytes after purgeMisplacedTasks movements
+        long byteLimit = (long) ((byteLimitFactor * config.getBytesMovedFullRebalance()) - moveAssignments.getBytesUsed());
```

Comment:
This is duplicate code from above. Can the useful bits be moved to a shared method?